### PR TITLE
usage example for free sprite

### PR DESCRIPTION
**Pull request by kay-kaushik**
# Description
This pull request features usage exaples for free_sprite(). The usage example is clear and concise to help the user understand how to implement the function.

# Features
+ clear and concise code
+ gifs
+ perfect example

# Changes
4 files and 1 folder have been added. 2 code, 1 text and a gif file.

# Benefits
These high quality usage examples give more context to the functions. This will help users reading the documentation to better understand the function. This will result in faster development.

# Testing
All the changes have been tested on Kartik Kaushik's system

# Testing Images

Screenshot 2024-08-23 150216

Screenshot 2024-08-23 150226

### CODE REVIEW

**Comment by breezy-codes** [comment]
## Code Issue in C++

When I try and run this code to test it doesn't work as there is an error for invalid bitmap. Can you add the png that was used in this code into the same location you placed the gif file so these codes can be tested.

image

## Code Issue in C#

All C# code needs to be done using top level only statements on the Splashkit site. Can you update this code too.

File: src/assets/usage-examples-code/sprites/free_sprite/free_sprite-3-free-sprite.cpp [hunk 1]
```diff
@@ -0,0 +1,42 @@
+/**
```
Comment (breezy-codes) [inline src/assets/usage-examples-code/sprites/free_sprite/free_sprite-3-free-sprite.cpp:1]: Remove this comment at the top, its not needed in the program code thats displayed.

File: src/assets/usage-examples-code/sprites/free_sprite/free_sprite-3-free-sprite.cs [hunk 2]
```diff
@@ -0,0 +1,45 @@
+/*
```
Comment (breezy-codes) [inline src/assets/usage-examples-code/sprites/free_sprite/free_sprite-3-free-sprite.cs:1]: Same with this comment

File: src/assets/usage-examples-code/sprites/free_sprite/free_sprite-3-free-sprite.cs [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    Usage Example: freeing a sprite
+*/
+
+using System;
+using SplashKitSDK;
+
+public class Program
```
Comment (breezy-codes) [inline src/assets/usage-examples-code/sprites/free_sprite/free_sprite-3-free-sprite.cs:8]: All C# programs on splashkit must be done using top level statements only. Can you update this to not use public class or static void.

File: src/assets/usage-examples-code/sprites/free_sprite/free_sprite-3-free-sprite.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+
+#include "splashkit.h"
+
+int main()
+{
+    // widow to draw the sprite on
+    window start = open_window("create_sprite", 600, 600);
+
+    // bitmap for creating a sprite
+    bitmap player = load_bitmap("playerBmp", "protSpriteSheet.png");
```
Comment (breezy-codes) [inline src/assets/usage-examples-code/sprites/free_sprite/free_sprite-3-free-sprite.cpp:13]: Add the png into the public/usage-examples-images-gifs/sprits

**Review by breezy-codes — CHANGES_REQUESTED** [review]
Ive added comments on the sections needing changes.
The comments at the top of each code aren't needed for displaying on the splashkit site.
All C# programs must use top level statements only on splashkit.

**Comment by breezy-codes** [comment]
The file you've added doesn't match up with the gif preview of the program running?

Otherwise the program runs correctly.

## C++ Code
Screenshot from 2024-08-31 11-17-27
Screenshot from 2024-08-31 11-17-29

## C# code
Screenshot from 2024-08-31 11-19-11
Screenshot from 2024-08-31 11-19-13

**Comment by breezy-codes** [comment]
Please update to add the correct image

File: src/assets/usage-examples-code/sprites/free_sprite/free_sprite-3-free-sprite.cpp (hunk 1 above)
Comment (kay-kaushik) [reply src/assets/usage-examples-code/sprites/free_sprite/free_sprite-3-free-sprite.cpp]: I have removed the comments

File: src/assets/usage-examples-code/sprites/free_sprite/free_sprite-3-free-sprite.cs (hunk 2 above)
Comment (kay-kaushik) [reply src/assets/usage-examples-code/sprites/free_sprite/free_sprite-3-free-sprite.cs]: Done

File: src/assets/usage-examples-code/sprites/free_sprite/free_sprite-3-free-sprite.cs (hunk 3 above)
Comment (kay-kaushik) [reply src/assets/usage-examples-code/sprites/free_sprite/free_sprite-3-free-sprite.cs]: I have made them top level

File: src/assets/usage-examples-code/sprites/free_sprite/free_sprite-3-free-sprite.cpp (hunk 4 above)
Comment (kay-kaushik) [reply src/assets/usage-examples-code/sprites/free_sprite/free_sprite-3-free-sprite.cpp]: I have addde the resource folder

**Comment by kay-kaushik** [comment]
> Please update to add the correct image

I have added the images

**Review by breezy-codes — APPROVED** [review]
All changes made.

**Pull request merged**